lib/shim/message-shim.js

```diff
@@ -649,132 +546,176 @@ function recordSubscribedConsume(nodule, properties, spec) {
       internal: false
     }
   })
+}
 
-  /**
-   * @param queue
-   * @param destinationName
-   */
-  function makeWrapConsumer(queue, destinationName) {
-    const msgDescDefaults = copy.shallow(spec)
-    if (destNameIsArg && destinationName != null) {
-      msgDescDefaults.destinationName = destinationName
+/**
+ * Helper function for recordSubscribedConsume
+ *
+ * @private
+ * @param {MessageShim} shim
+ *  The shim used to wrap the subscribe function
+ * @param {MessageSubscribeSpec} spec
+ *  The specification for this subscription method's interface.
+ * @param {Function} subscribe
+ *  The original subscribe function to wrap.
+ * @returns {Function} the wrapped subsribe function
+ */
+function _makeWrappedSubscribe(shim, spec, subscribe) {
+  return function wrappedSubscribe() {
+    const args = shim.argsToArray.apply(shim, arguments)
+    const queueIdx = shim.normalizeIndex(args.length, spec.queue)
+    const consumerIdx = shim.normalizeIndex(args.length, spec.consumer)
+    const queue = queueIdx === null ? null : args[queueIdx]
+    let destName = null
+
+    // create a copy so we don't modify the original passed in by the
+    // user
+    spec = copy.shallow(spec)
+    if (shim.isNumber(spec.destinationName)) {
+      const destNameIdx = shim.normalizeIndex(args.length, spec.destinationName)
+      if (destNameIdx !== null) {
+        destName = args[destNameIdx]
+      }
+      if (destName != null) {
+        spec.destinationName = destName
+      }
     }
+
     if (queue != null) {
-      msgDescDefaults.queue = queue
+      spec.queue = queue
     }
 
-    return function wrapConsumer(shim, consumer, cName) {
-      if (!shim.isFunction(consumer)) {
-        return consumer
-      }
+    if (consumerIdx !== null) {
+      args[consumerIdx] = shim.wrap(args[consumerIdx], function wrapConsumer(shim, consumer, name) {
```

Comment:
interesting that you were able to cut out a lot of code by checking things above and then just returning this simple function

Reply:
I accidentally cut too much 'cause I got confused by the callback of the callback. I just put a bit back in.